### PR TITLE
feat: move to typescript code generation

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

File: synth.py [hunk 1]
```diff
@@ -21,34 +21,42 @@
...
-    'src/v1/device_manager_client.js',
-    f'(listDeviceRegistriesStream\(.*\n\s+options = .*\n)(\n\s+return)',
-    '''\g<1>options.otherArgs = options.otherArgs || {};
-    options.otherArgs.headers = options.otherArgs.headers || {};
-    options.otherArgs.headers[
-      'x-goog-request-params'
-    ] = gax.routingHeader.fromParams({
-      parent: request.parent,
-    });\n\g<2>''')
+# s.replace(
```
Comment (JustinBeckwith) [inline synth.py]: If we don't need this anymore, let's remove it.  Hard to say what it was originally doing ....

Comment (alexander-fenster) [reply synth.py]: The issue it refers to is closed long time ago. We should just remove this comment.

File: webpack.config.js [hunk 2]
```diff
@@ -12,33 +12,51 @@
...
 
+const path = require('path');
+
 module.exports = {
-  entry: './src/browser.js',
+  entry: './src/index.ts',
   output: {
-    library: 'iot',
-    filename: './iot.js',
+    library: 'DeviceManager',
```
Comment (alexander-fenster) [inline webpack.config.js]: This is breaking! (even though nobody uses it)

Let's use our cool new option `--main-service iot` here.

File: synth.py (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply synth.py]: understood, removed it, thanks

**Pull request merged**